rpcs3/Emu/RSX/VK/VKFragmentProgram.cpp

```diff
@@ -207,6 +213,19 @@ void VKFragmentDecompilerThread::insertMainStart(std::stringstream & OS)
 				vk::insert_fog_declaration(OS, m_prog.fog_equation);
 				return;
 			}
+			
+			if (m_prog.front_back_color_enabled)
+			{
+				if (PI.name == "spec_color" && m_prog.back_color_specular_output)
+				{
+					OS << "	vec4 spec_color = gl_FrontFacing ? vec4(front_spec_color) : vec4(spec_color);\n";
+				}
+				if (PI.name == "diff_color" && m_prog.back_color_diffuse_output)
+				{
+					OS << "	vec4 diff_color = gl_FrontFacing ? vec4(front_diff_color) : vec4(diff_color);\n";
+				}
+				return;
```

Comment:
@raven02 So you added the diff_color, but.. is it actually consumed by the shader? If it was and the declaration was missing you'd have gotten a compilation error. This means that this code will just be removed by the glsl compiler as dead code.